include/rdycore.h.in

```diff
@@ -47,20 +47,42 @@ PETSC_EXTERN PetscErrorCode RDyAdvance(RDy);
 PETSC_EXTERN PetscErrorCode RDyGetCouplingInterval(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDySetCouplingInterval(RDy, PetscReal);
+
 PETSC_EXTERN PetscErrorCode RDyGetNumLocalCells(RDy, PetscInt*);
 PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryConditions(RDy, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryEdges(RDy, PetscInt, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDyGetBoundaryConditionFlowType(RDy, PetscInt, PetscInt*);
-PETSC_EXTERN PetscErrorCode RDySetDirichletBoundaryValues(RDy, PetscInt, PetscInt, PetscInt, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetHeight(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetXVelocity(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDyGetYVelocity(RDy, PetscReal*);
-PETSC_EXTERN PetscErrorCode RDySetWaterSource(RDy, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetNumBoundaryEdges(RDy, const PetscInt, PetscInt*);
+PETSC_EXTERN PetscErrorCode RDyGetBoundaryConditionFlowType(RDy, const PetscInt, PetscInt*);
+
+PETSC_EXTERN PetscErrorCode RDyGetHeightOfLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetXMomentumOfLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetYMomentumOfLocalCell(RDy, const PetscInt, PetscReal*);
+
+PETSC_EXTERN PetscErrorCode RDyGetXCentroidOfLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetYCentroidOfLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetZCentroidOfLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetNatIDOfLocalCell(RDy, const PetscInt, PetscInt*);
+
+PETSC_EXTERN PetscErrorCode RDyGetXCentroidOfBoundaryEdge(RDy, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetYCentroidOfBoundaryEdge(RDy, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetZCentroidOfBoundaryEdge(RDy, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetXCentroidOfBoundaryCell(RDy, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetYCentroidOfBoundaryCell(RDy, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetZCentroidOfBoundaryCell(RDy, const PetscInt, const PetscInt, PetscReal*);
+
+PETSC_EXTERN PetscErrorCode RDyGetNaturalIDOfBoundaryCell(RDy, const PetscInt, const PetscInt, PetscInt*);
+
+PETSC_EXTERN PetscErrorCode RDySetCouplingInterval(RDy, PetscReal);
+
+PETSC_EXTERN PetscErrorCode RDySetDirichletBoundaryValues(RDy, const PetscInt, const PetscInt, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDySetWaterSourceForLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDySetXMomentumSourceForLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDySetYMomentumSourceForLocalCell(RDy, const PetscInt, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDySetManningNForLocalCell(RDy, const PetscInt, PetscReal*);
```

Comment:
We can articulate the size of the array for the Manning coefficients by using named parameters with defined array sizes like this:

```
PETSC_EXTERN PetscErrorCode RDySetManningNForLocalCell(RDy rdy, const PetscInt size, PetscReal n_values[size]);
```

That makes it clear to the caller how the array should be sized to receive the coefficients.